src/osd/ReplicatedPG.cc

```diff
@@ -11777,17 +11793,18 @@ void ReplicatedPG::replace_cached_attrs(
   }
 }
 
-void ReplicatedPG::setattr_maybe_cache(
+void ReplicatedPG::setattrs_maybe_cache(
   ObjectContextRef obc,
   OpContext *op,
   PGBackend::PGTransaction *t,
-  const string &key,
-  bufferlist &val)
+  map<string, bufferlist> &attrs)
 {
   if (pool.info.require_rollback()) {
-    op->pending_attrs[obc][key] = val;
+    for (map<string, bufferlist>::iterator it = attrs.begin();
+      it != attrs.end(); it++)
+    op->pending_attrs[obc][it->first] = it->second;
   }
-  t->setattr(obc->obs.oi.soid, key, val);
+  t->setattrs(obc->obs.oi.soid, attrs);
```

Comment:
perhaps keep setattr_maybe_cache and *also* have a setattrs_maybe_cache?  @athanatos?